### PR TITLE
Considere an empty query search as a placeholder search

### LINKED ISSUE #856: Placeholder search when query is empty does not work
**Describe the bug**
When using the placeholder search you can either:
1. Not give the parameter `q`: In which case the placeholder search works as expected
2. Give an empry parameter `q`: in which case **no documents are returned**

**To Reproduce**
1. Create an index and add documents to it
2. Make a search with an empty query parameter
3. No documents are returned

**Expected behavior**
As it is a placeholder search, I expected it to work with an empty string. I'm not sure this is a bug as it is maybe what you have decided. I could not find a discussion about it on the related tracking issue. 

**Screenshots**

```bash
➜ curl -X GET 'http://localhost:7700/indexes/movies/search?q='  | jq
{
  "hits": [],
  "offset": 0,
  "limit": 20,
  "nbHits": 0,
  "exhaustiveNbHits": false,
  "processingTimeMs": 3,
  "query": ""
}
```

**Tests should be added**
- [ ] empty `q` with expected behavior
```rust
let query = json! ({
        "q": ""
    });
```

### PR DESCRIPTION
**Pull request by qdequele**
Fix #856; Relative tracking issue: #729

### CODE REVIEW

**Comment by MarinPostma** [comment]
Nice rust-fmt haha

**Comment by qdequele** [comment]
I now use VSCode with a [rust-analyzer](https://github.com/rust-analyzer/rust-analyzer) 😉

**Comment by Kerollmops** [comment]
Could you please split the rustfmt thing from the core change you've done?

It would be easier to review.

File: meilisearch-http/src/routes/search.rs [hunk 1]
```diff
@@ -105,8 +105,13 @@ impl SearchQuery {
             .schema(&reader)?
             .ok_or(Error::internal("Impossible to retrieve the schema"))?;
 
-        let mut search_builder = index.new_search(self.q.clone());
+        let query = self
+            .q
+            .clone()
+            .map(|q| if q.is_empty() { None } else { Some(q) })
+            .flatten();
```
Comment (Kerollmops) [inline meilisearch-http/src/routes/search.rs:115]: https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then

**Review by Kerollmops — CHANGES_REQUESTED** [review]

File: meilisearch-http/src/routes/search.rs (hunk 1 above)
Comment (qdequele) [reply meilisearch-http/src/routes/search.rs]: Thanks, it's fixed!

**Review by Kerollmops — APPROVED** [review]

**Comment by Kerollmops** [comment]
@qdequele Whoops there is a merge conflict, could you rebase, please?

**Review by MarinPostma — APPROVED** [review]
bors r+

**Pull request merged**